aea/helpers/dialogue/base.py

```diff
@@ -412,90 +395,145 @@ def last_message(self) -> Optional[Message]:
 
         return last_message
 
-    def get_message(self, message_id_to_find: int) -> Optional[Message]:
+    @property
+    def is_empty(self) -> bool:
+        """
+        Check whether the dialogue is empty.
+
+        :return: True if empty, False otherwise
+        """
+        return len(self._outgoing_messages) == 0 and len(self._incoming_messages) == 0
+
+    def counterparty_from_message(self, message: Message) -> Address:
+        if self.is_message_by_self(message):
+            counterparty = message.to
+        else:
+            counterparty = message.sender
+
+        return counterparty
+
+    def is_message_by_self(self, message: Message) -> bool:
+        """
+        Check whether the message is by this agent or not.
+
+        :param message: the message
+        :return: True if message is by this agent, False otherwise
+        """
+        return message.sender == self.agent_address
+
+    def is_message_by_other(self, message: Message) -> bool:
+        """
+        Check whether the message is by the counterparty agent in this dialogue or not.
+
+        :param message: the message
+        :return: True if message is by the counterparty agent in this dialogue, False otherwise
+        """
+        return not self.is_message_by_self(message)
+
+    def get_message(self, message_id: int) -> Message:
         """
         Get the message whose id is 'message_id'.
 
-        :param message_id_to_find: the id of the message
+        :param message_id: the id of the message
         :return: the message if it exists, None otherwise
         """
-        result = None  # type: Optional[Message]
+        assert not self.is_empty, "This dialogue is empty"
+        assert self.has_message_id(
+            message_id
+        ), "A message with this id does not exists. Expected a message id between 1 and {}. Found {}.".format(
+            self.last_message.message_id, message_id
+        )
+
         list_of_all_messages = self._outgoing_messages + self._incoming_messages
         for message in list_of_all_messages:
-            if message.message_id == message_id_to_find:
-                result = message
-                break
+            if message.message_id == message_id:
+                return message
```

Comment:
You could have another method which wraps this section like so:
````
def try_get_method(self, message_id: int) -> Optional[Message]:
    """
    ...
    """
    result = None  # type: Optional[Message]
    list_of_all_messages = self._outgoing_messages + self._incoming_messages
    for message in list_of_all_messages:
          if message.message_id == message_id_to_find:
                 result = message
                 break
      return  result
```
(i.e. the old `get_message`) and then you use this in the new `get_message`; that way you can then add a line in the new `get_message`:
```
message = try_get_message(message_id)
assert message is not None, "Message not present"
return message
```

Reply:
I added this, but I think this adds unnecessary repetition to the module and confusion in usage (which method to use) while adding very minor benefits.

Reply:
There is no repetition; it's two methods offering different behaviours on accessing the message from its id.